File: AiAgentGame2/langgraph-studio/src/types/project.ts
export type ProjectStatus='draft'|'running'|'paused'|'completed'|'failed'

export type PhaseNumber=1|2|3

export interface Project{
 id:string
 name:string
 description?:string
 concept?:GameConcept
 status:ProjectStatus
 currentPhase:PhaseNumber
 state?:Record<string,unknown>
 config?:ProjectConfig
 createdAt:string
 updatedAt:string
}

export interface GameConcept{
 description:string
 platform:'web'|'desktop'|'mobile'
 scope:'mvp'|'full'
 genre?:string
 targetAudience?:string
}

export interface ProjectConfig{
 llmProvider?:'claude'|'gpt4'|'mock'
 maxTokensPerAgent?:number
 enableAssetGeneration?:boolean
 enableAutoApproval?:boolean
}

export interface CreateProjectInput{
 name:string
 description?:string
 concept:GameConcept
 config?:ProjectConfig
}

export interface ProjectMetrics{
 projectId:string
 totalTokensUsed:number
 estimatedTotalTokens:number
 elapsedTimeSeconds:number
 estimatedRemainingSeconds:number
 estimatedEndTime:string|null
 completedTasks:number
 totalTasks:number
 progressPercent:number
 currentPhase:PhaseNumber
 phaseName:string
}

File: backend/schemas/project.py
from typing import Optional, Dict, Any
from datetime import datetime
from .base import BaseSchema


class ProjectSchema(BaseSchema):
    id: str
    name: str
    description: Optional[str] = None
    concept: Optional[Dict[str, Any]] = None
    status: str
    current_phase: int
    state: Optional[Dict[str, Any]] = None
    config: Optional[Dict[str, Any]] = None
    ai_services: Optional[Dict[str, Any]] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    output_settings: Optional[Dict[str, Any]] = None


class ProjectCreateSchema(BaseSchema):
    name: str
    description: Optional[str] = None
    concept: Optional[Dict[str, Any]] = None
    config: Optional[Dict[str, Any]] = None
    ai_services: Optional[Dict[str, Any]] = None


class ProjectUpdateSchema(BaseSchema):
    name: Optional[str] = None
    description: Optional[str] = None
    concept: Optional[Dict[str, Any]] = None
    status: Optional[str] = None
    current_phase: Optional[int] = None
    state: Optional[Dict[str, Any]] = None
    config: Optional[Dict[str, Any]] = None
    ai_services: Optional[Dict[str, Any]] = None
    output_settings: Optional[Dict[str, Any]] = None
